From c0430ac05851d935ded77661b4ec5163286f8b5d Mon Sep 17 00:00:00 2001
From: Igor Derzhavets <igor.derzhavets@compulab.co.il>
Date: Tue, 30 Jul 2019 15:11:02 +0300
Subject: [PATCH 12/23] Set FlexSPI speed to 133MHz

Signed-off-by: Igor Derzhavets <igor.derzhavets@compulab.co.il>
---
 include/configs/cl-som-imx8x.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/cl-som-imx8x.h b/include/configs/cl-som-imx8x.h
index 45d7581bc4..153b0943ea 100644
--- a/include/configs/cl-som-imx8x.h
+++ b/include/configs/cl-som-imx8x.h
@@ -383,7 +383,7 @@
 #ifdef CONFIG_FSL_FSPI
 #define CONFIG_SF_DEFAULT_BUS		0
 #define CONFIG_SF_DEFAULT_CS		0
-#define CONFIG_SF_DEFAULT_SPEED	40000000
+#define CONFIG_SF_DEFAULT_SPEED	133000000
 #define CONFIG_SF_DEFAULT_MODE		SPI_MODE_3
 #define FSL_FSPI_FLASH_SIZE		SZ_2M
 #define FSL_FSPI_FLASH_NUM		1
-- 
2.11.0

